helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PerInstanceAccessor.java

```diff
@@ -185,6 +200,137 @@ public Response isInstanceStoppable(
     return OK(OBJECT_MAPPER.writeValueAsString(stoppableCheck));
   }
 
+  /**
+   * Performs health checks, user designed operation check and execution for take an instance.
+   *
+   * @param jsonContent json payload
+   * @param clusterId cluster id
+   * @param instanceName Instance name to be checked
+   * @return json response representing if queried instance is stoppable
+   * @throws IOException if there is any IO/network error
+   */
+  @ResponseMetered(name = HttpConstants.WRITE_REQUEST)
+  @Timed(name = HttpConstants.WRITE_REQUEST)
+  @POST
+  @Path("takeInstance")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response takeSingleInstance(
+      String jsonContent,
+      @PathParam("clusterId") String clusterId,
+      @PathParam("instanceName") String instanceName){
+
+    try {
+      JsonNode node = null;
+      if (jsonContent.length() != 0) {
+        node = OBJECT_MAPPER.readTree(jsonContent);
+      }
+      if (node == null) {
+        return badRequest("Invalid input for content : " + jsonContent);
+      }
+      MaintenanceOpInputFields inputFields = readMaintenanceInputFromJson(node);
+
+      MaintenanceManagementService maintenanceManagementService =
+          new MaintenanceManagementService((ZKHelixDataAccessor) getDataAccssor(clusterId),
+              getConfigAccessor(), inputFields.skipZKRead, inputFields.continueOnFailures,
+              getNamespace());
+
+      return JSONRepresentation(maintenanceManagementService
+          .takeInstance(clusterId, instanceName, inputFields.healthChecks,
+              inputFields.healthCheckConfig,
+              inputFields.operations,
+              inputFields.operationConfig, true));
+    } catch (Exception e) {
+      LOG.error("Failed to takeInstances:", e);
+      return badRequest("Failed to takeInstances: " + e.getMessage());
+    }
+  }
+
+  /**
+   * Performs health checks, user designed operation check and execution for free an instance.
+   *
+   * @param jsonContent json payload
+   * @param clusterId cluster id
+   * @param instanceName Instance name to be checked
+   * @return json response representing if queried instance is stoppable
+   * @throws IOException if there is any IO/network error
+   */
+  @ResponseMetered(name = HttpConstants.WRITE_REQUEST)
+  @Timed(name = HttpConstants.WRITE_REQUEST)
+  @POST
+  @Path("freeInstance")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response freeSingleInstance(
+      String jsonContent,
+      @PathParam("clusterId") String clusterId,
+      @PathParam("instanceName") String instanceName){
+
+    try {
+      JsonNode node = null;
+      if (jsonContent.length() != 0) {
+        node = OBJECT_MAPPER.readTree(jsonContent);
+      }
+      if (node == null) {
+        return badRequest("Invalid input for content : " + jsonContent);
+      }
+      MaintenanceOpInputFields inputFields = readMaintenanceInputFromJson(node);
```

Comment:
This is dup code with previous function. Let's wrap it into readMaintenanceInputFromJson function?